includes/widgets/traits/button-trait.php

```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
+						'icon' => "eicon-h-align-{$start}",
+					],
+					'row-reverse' => [
+						'title' => esc_html__( 'End', 'elementor' ),
+						'icon' => "eicon-h-align-{$end}",
+					],
+					'column' => [
+						'title' => esc_html__( 'Above', 'elementor' ),
+						'icon' => "eicon-v-align-top",
+					],
+					'column-reverse' => [
+						'title' => esc_html__( 'Below', 'elementor' ),
+						'icon' => "eicon-v-align-bottom",
+					],
+				],
+				'selectors_dictionary' => [
+					'left' => is_rtl() ? 'row-reverse' : 'row',
+					'right' => is_rtl() ? 'row' : 'row-reverse',
+					'above' => is_rtl() ? 'column-reverse' : 'column',
+					'below' => is_rtl() ? 'column' : 'column-reverse',
+				],
+				'selectors' => [
+					'{{WRAPPER}} .elementor-button-content-wrapper' => 'flex-direction: {{VALUE}};',
+				],
+				'condition' => array_merge(
+					$args['section_condition'],
+					[
+						'text!' => '',
+						'selected_icon[value]!' => '',
+					]
+				),
+			]
+		);
+
+		$this->add_responsive_control( 
+			'icon_size', 
+			[
+				'label' => esc_html__( 'Size', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'range' => [
+					'px' => [
+						'max' => 100,
+					],
+					'em' => [
+						'max' => 10,
+					],
+					'rem' => [
+						'max' => 10,
+					],
+				],
+				'size_units' => [ 'px', 'em', 'rem', 'vw', 'custom' ],
+				'selectors' => [
+					'{{WRAPPER}}' => '--n-button-icon-size: {{SIZE}}{{UNIT}}',
+				],
+		] );
+
+		$this->add_control(
+			'icon_indent',
+			[
+				'label' => esc_html__( 'Icon Spacing', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', 'em', 'rem', 'custom' ],
+				'range' => [
+					'px' => [
+						'max' => 50,
+					],
+					'em' => [
+						'max' => 5,
+					],
+					'rem' => [
+						'max' => 5,
+					],
+				],
+				'selectors' => [
+					'{{WRAPPER}} .elementor-button .elementor-button-content-wrapper' => 'gap: {{SIZE}}{{UNIT}};',
+				],
+				'condition' => array_merge(
+					$args['section_condition'],
+					[
+						'text!' => '',
+						'selected_icon[value]!' => '',
+					]
+				),
+			]
+		);
+
+		//start the color tabs
+		$this->start_controls_tabs( 'icons_button_style', [
+			'condition' => $args['section_condition'],
+		] );
+				//when the icon is on normal
+				$this->start_controls_tab(
+					'icon_button_normal',
+					[
+						'label' => esc_html__( 'Normal', 'elementor' ),
+						'condition' => $args['section_condition'],
+						]
+					);
+					
+				$this->add_control( 'icon_color', [
+						'label' => esc_html__( 'Color', 'elementor' ),
+						'type' => Controls_Manager::COLOR,
+						'selectors' => [
+							'{{WRAPPER}}' => '--button-icon-color: {{VALUE}};',
+						],
+					] );
+						
+				$this->end_controls_tab();
+						
+				//when the icon is on hover
+				$this->start_controls_tab(
+						'icon_button_hover',
+						[
+							'label' => esc_html__( 'Hover', 'elementor' ),
+							'condition' => $args['section_condition'],
```

Comment:
Do we need this?
What is this adding?

Reply:
for some reason the size does not work when I remove it

Reply:
Let's leave it for now. We can check it later.